spec/puppet-syntax/manifests_spec.rb

```diff
@@ -78,19 +114,34 @@
         context 'Puppet >= 3.2' do
           it 'should pass with future option set to true on future manifest' do
             files = fixture_manifests(['future_syntax.pp'])
-            res = subject.check(files)
+            output, has_errors = subject.check(files)
 
-            expect(res.size).to eq(0)
+            expect(output).to eq([])
+            expect(has_errors).to eq(false)
+          end
+        end
+        context 'Puppet >= 3.7' do
+          # Certain elements of the future parser weren't added until 3.7
+          if Puppet::Util::Package.versioncmp(Puppet.version, '3.7') >= 0
+            it 'should fail on what were deprecation notices in the < 4.0 parser' do
```

Comment:
The description for this `it` block should also be updated since we're not excluding version 4.0 in this test.